### ISSUE TITLE
Test updates for NCBI BLAST DB v5

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#2863]
See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=959587

```
> test_NCBI_BLAST_tools ... /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ blastn and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ blastp and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -ipglist,-negative_ipglist,-negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ blastx and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -ipglist,-negative_ipglist,-negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ deltablast and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ makeblastdb and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -blastdb_version)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ psiblast and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -ipglist,-negative_ipglist,-negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ rpsblast and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -sorthits,-sorthsps,-subject_besthit)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ rpstblastn and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -sorthits,-sorthsps)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ tblastn and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> /<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py:398: UserWarning: NCBI BLAST+ tblastx and Biopython out sync. Please update Biopython, or report this issue if you are already using the latest version. (Extra args: ; Missing: -negative_taxidlist,-negative_taxids,-sorthits,-sorthsps,-subject_besthit,-taxidlist,-taxids)
>   warnings.warn("NCBI BLAST+ %s and Biopython out sync. Please "
> FAIL
```

```
> ======================================================================
> FAIL: test_fasta_db_nucl (test_NCBI_BLAST_tools.BlastDB)
> Test makeblastdb wrapper with nucleotide database.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py", line 248, in test_fasta_db_nucl
>     self.assertTrue(os.path.isfile("GenBank/NC_005816.fna.nsd"))
> AssertionError: False is not true
> 
> ======================================================================
> FAIL: test_fasta_db_prot (test_NCBI_BLAST_tools.BlastDB)
> Test makeblastdb wrapper with protein database.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py", line 213, in test_fasta_db_prot
>     self.assertTrue(os.path.isfile("GenBank/NC_005816.faa.psd"))
> AssertionError: False is not true
```

@tillea wrote:

> per https://www.ncbi.nlm.nih.gov/books/NBK131777/, the upgrade to
2.10.0 changed the default database version (controlled by makeblastdb's
-blastdb_version flag) from 4 to 5, causing some output files to change
names (and formats):
> 
> *.nsd -> *.nnd
> *.nsi -> *.nni
> *.psd -> *.pnd
> *.psi -> *.pni

Related to this, there are new switches to add to the command line wrappers (shown as warnings from the tests above)

**Comment by peterjc** [comment]
Test failure due to new extensions fixed in #2801 

Closing issue, see #2184 for the changes to the command line options.

**Comment by peterjc** [comment]
@tillea reported by email that even with the patch it fails on 32bit systems:

```
 ======================================================================
> FAIL: test_fasta_db_nucl (test_NCBI_BLAST_tools.BlastDB)
> Test makeblastdb wrapper with nucleotide database.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py", line 249, in test_fasta_db_nucl
>     self.assertTrue(os.path.isfile("GenBank/NC_005816.fna.nhd"))
> AssertionError: False is not true
>
> ======================================================================
> FAIL: test_fasta_db_prot (test_NCBI_BLAST_tools.BlastDB)
> Test makeblastdb wrapper with protein database.
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.8/build/Tests/test_NCBI_BLAST_tools.py", line 208, in test_fasta_db_prot
>     self.assertTrue(os.path.isfile("GenBank/NC_005816.faa.phd"))
> AssertionError: False is not true
```

**Comment by mdehoon** [comment]
@chebizarro @peterjc  Is this still a blocking issue for release 1.77?

**Comment by peterjc** [comment]
Possibly - I suspect it is a Debian/Blast issue though.

**Comment by mdehoon** [comment]
Then does it make sense to wait for this? Especially since we are considering to drop `Bio.Application`.

**Comment by peterjc** [comment]
It works on the major CPUs anyway - seems low risk to proceed with our release.

**Comment by tillea** [comment]
On Sat, May 23, 2020 at 08:09:56AM -0700, Peter Cock wrote:
> It works on the major CPUs anyway - seems low risk to proceed with our release.
I'd recommend to proceed which also saves some patches on Debian side.  I did not found the time to follow your hints yet but I can perfectly try to hunt this down on the next version.

**Comment by chebizarro** [comment]
I built a Docker container based on the i386/ubuntu 18.04 image and can't seem to replicate the problem using the i386 build of `makeblastdb` 2.6.0. I can try different base image/blast+ combinations to try and pin down the error but that won't be until next week.

**Comment by peterjc** [comment]
Thanks Chris, that is reassuring - less likely one of our main target demographic will hit this.

**Comment by peterjc** [comment]
This would be solved by #3344

**Comment by peterjc** [comment]
Clarification: The warning about the BLAST+ wrappers being out of sync would be fixed by dropping them.

The error about V5 databases was already fixed, as as the ``-blastdb_version`` flag.

Closing.
